### PR TITLE
fix: subject fixed failure to propgate extension rules present in standard

### PR DESCRIPTION
**Pull request by 6XGate**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[x] Bug fix
[ ] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**

Ensured that all `@typescript-eslint` rules are propagated if they appear in the Standard JS configuration.

**Which issue (if any) does this pull request address?**

#582 Some upstream standard rules are not properly propagated to their @typescript-eslint version

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

**Review by Divlo — APPROVED** [review]
Looks good to me! :+1: 
But I don't know much of this codebase, so I'll let @mightyiam decide to approve and merge this PR.

Thanks for your contribution! @6XGate

File: package.json [hunk 1]
```diff
@@ -55,24 +55,24 @@
...
+    "typescript": ">=3.9"
   },
   "devDependencies": {
     "@commitlint/cli": "12.1.4",
     "@commitlint/config-conventional": "12.1.4",
     "@commitlint/travis-cli": "12.1.4",
     "@types/eslint": "7.2.13",
     "@types/node": "14.17.3",
-    "@typescript-eslint/eslint-plugin": "4.0.1",
+    "@typescript-eslint/eslint-plugin": "4.15.0",
```
Comment (mightyiam) [inline package.json:75]: I'd like dependency upgrades in a separate PR, even if they are necessary for this one, please.

**Review by mightyiam — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Divlo) [reply package.json:75]: PR available there: #592

Comment (6XGate) [reply package.json:75]: #592 failed checks due to one of the dependency constraints changing in `standard` that this code base originally take into account. I fixed it here, so hope that won't be an issue.

Comment (Divlo) [reply package.json:75]: Actually, it will be fixed with #590.

**Comment by rostislav-simonik** [comment]
@6XGate sorry for not reviewing your PR until now, if we review it, will you proceed with it?

**Comment by 6XGate** [comment]
proceed with it? I'm not sure what you mean by that.
It likely will need to corrections due to recently landed PRs.
It was meant to better keep in sync with standard by sync all know (at the time) shared/replaced plain ESLint rules with their TypeScript counter part

**Comment by mightyiam** [comment]
@6XGate yes. If we review it will you follow through with it with it? It's been a while. That's why we ask.

**Comment by 6XGate** [comment]
Sure, I'll give a try at it again soon.

**Comment by mightyiam** [comment]
I'm happy to hear that, @6XGate . But we haven't reviewed it yet, so don't spend time yet. Now that I know you're willing to go through with it, we'll review it for you. Not before August though.

**Pull request closed without merging**